### PR TITLE
CSPL-720: Adding support to configure a custom serviceaccount per CRD

### PR DESCRIPTION
**Pull request by akondur**
- Added support to configure custom service account per CRD. 
- Added util apis for service accounts
- Added UT for code coverage & config testing
- Changed documentation as necessary

### CODE REVIEW

File: pkg/splunk/controller/serviceaccount.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2018-2020 Splunk Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/controller/serviceaccount.go:1]: Resolved in a different [PR](https://github.com/splunk/splunk-operator/pull/228)

File: pkg/splunk/controller/serviceaccount_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2018-2020 Splunk Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/controller/serviceaccount_test.go:1]: Same as above

**Review by smohan-splunk — APPROVED** [review]

File: deploy/crds/enterprise.splunk.com_clustermasters_crd.yaml [hunk 3]
```diff
@@ -752,6 +752,10 @@ spec:
               description: Name of Scheduler to use for pod placement (defaults to
                 “default-scheduler”)
               type: string
+            serviceAccount:
```
Comment (jryb) [inline deploy/crds/enterprise.splunk.com_clustermasters_crd.yaml:755]: Does the `serviceAccount` default to anything if not set?  Should we document that here?

File: pkg/splunk/controller/serviceaccount.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+// ApplyServiceAccount creates or updates a Kubernetes serviceAccount
+func ApplyServiceAccount(client splcommon.ControllerClient, serviceAccount *corev1.ServiceAccount) error {
+	namespacedName := types.NamespacedName{Namespace: serviceAccount.GetNamespace(), Name: serviceAccount.GetName()}
+	var current corev1.ServiceAccount
+
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err == nil {
+		if !reflect.DeepEqual(serviceAccount, &current) {
```
Comment (jryb) [inline pkg/splunk/controller/serviceaccount.go:35]: Do we need to throw an error or print a log here if the Apply fails?

File: pkg/splunk/enterprise/configuration.go [hunk 5]
```diff
@@ -441,6 +441,16 @@ func getSplunkStatefulSet(client splcommon.ControllerClient, cr splcommon.MetaOb
 		return statefulSet, err
 	}
 
+	// add serviceaccount if configured
+	if spec.ServiceAccount != "" {
+		namespacedName := types.NamespacedName{Namespace: statefulSet.GetNamespace(), Name: spec.ServiceAccount}
+		_, err := splctrl.GetServiceAccount(client, namespacedName)
+		if err == nil {
+			// serviceAccount exists
```
Comment (jryb) [inline pkg/splunk/enterprise/configuration.go:449]: Do we want to bail here if we have an error?

File: deploy/crds/enterprise.splunk.com_clustermasters_crd.yaml (hunk 3 above)
Comment (akondur) [reply deploy/crds/enterprise.splunk.com_clustermasters_crd.yaml:755]: If no custom serviceaccount is attached to a pod, a default service account assigned to the cluster is used by the pod. Let me document that.

File: pkg/splunk/controller/serviceaccount.go (hunk 4 above)
Comment (akondur) [reply pkg/splunk/controller/serviceaccount.go:35]: `If err != nil`  it means that the object was found and hence we don't need to error out/print log?

File: pkg/splunk/enterprise/configuration.go (hunk 5 above)
Comment (akondur) [reply pkg/splunk/enterprise/configuration.go:449]: **GetServiceAccount** throws an error as per this [log](https://github.com/splunk/splunk-operator/pull/220/files#diff-b0b10241f20d210433da861923f8978a724ec51c2d73c668b3163951eb3dccedR53).

File: pkg/splunk/controller/serviceaccount.go (hunk 4 above)
Comment (jryb) [reply pkg/splunk/controller/serviceaccount.go:35]: Ok so the cases here are: The context doesn't exists and we create it.  Or it exists and we check if its the same and if not, call `UpdateResource`.  If there is an error in either its returned to the caller.  Do we want a log stateing we're updating it?

Comment (akondur) [reply pkg/splunk/controller/serviceaccount.go:35]: Got it. Making this change as well now.

**Review by jryb — APPROVED** [review]

**Pull request merged**